### ISSUE TITLE
Allow redeployment of unchanged contract

### ISSUE THREAD

**Issue by monkybrain** [SuperblocksHQ/superblocks-lab#400]
Hey! It would be great if you could redeploy contracts that haven't been changed. Quite frequently during the development process I just want to reset the state of the contract, i.e. deploy the same contract to a different address. Currently it seems like you have to edit the contract to force the IDE to recompile and deploy a new contract. Or perhaps I've missed something?

**Comment by javier-tarazaga** [comment]
Hi @monkybrain! 

@filippsen might give you a more detail explanation of this, but the initial idea was to avoid actually re-deployments of the same contract when actually nothing has actually changed.  But I can see the point of wanting to deploy the same contract to a different address.

**Comment by filippsen** [comment]
@javier-tarazaga Correct. The main reason for that behavior was to not repeat the exact same work, as mentioned. Since deployment data is saved as part of the project, the system detects the files already exist and haven't changed since last time. Also, when you _Deploy_ to an external network you are able to _Interact_ and _Preview_ on later sessions because of the saved state (continue later without having to redeploy).

An idea to consider could be to add the option to always force redeployment when triggering any _Deploy_ action. Extra notes: a) consider defaults to `no`, `false` or `off`; b) might be healthy to not allow this new setting when targeting non-test networks (as of now, Mainnet) or show an extra warning message before doing so.

**Comment by monkybrain** [comment]
@filippsen I see your point and makes sense to me too. Perhaps a "No changes detected. Want to redeploy anyway?" dialog would do the trick? (perhaps even with a "don't ask me again" option, at least for test environments, on a per project basis)
